genai-function-calling/vercel-ai/index.js

```diff
@@ -42,20 +43,35 @@ const getLatestElasticsearchVersion = tool({
     },
 });
 
-async function main() {
+const tools = {get_latest_elasticsearch_version: getLatestElasticsearchVersion};
+
+/**
+ * Runs the agent with the given tools.
+ *
+ * @param {ToolSet} tools - The tools the LLM can use to answer the question.
+ */
+async function runAgent(tools = tools) {
```

Comment:
IIUC a parameter is always passed currently - maybe remove the default to prevent accidentally using wrong tools when changing something

Reply:
good idea